crash-lib/src/jvmMain/kotlin/co/electriccoin/zcash/crash/ExceptionPath.kt

```diff
@@ -0,0 +1,33 @@
+package co.electriccoin.zcash.crash
+
+import co.electriccoin.zcash.spackle.io.canWriteSuspend
+import co.electriccoin.zcash.spackle.io.existsSuspend
+import co.electriccoin.zcash.spackle.io.isDirectorySuspend
+import co.electriccoin.zcash.spackle.io.mkdirsSuspend
+import java.io.File
+import java.util.UUID
+
+object ExceptionPath {
+    const val LOG_DIRECTORY_NAME = "log" // $NON-NLS-1$
+    const val EXCEPTION_DIRECTORY_NAME = "exception" // $NON-NLS-1$
+    const val SEPARATOR = "|"
+
+    fun newExceptionFileName(exception: ReportableException, uuid: UUID = UUID.randomUUID()) =
+        "${exception.time.epochSeconds}$SEPARATOR$uuid$SEPARATOR${exception.javaClass.simpleName}.txt"
+
+    suspend fun validateDir(path: File) {
+        if (!path.existsSuspend()) {
+            if (!path.mkdirsSuspend()) {
+                throw IllegalArgumentException("Directories couldn't be created")
```

Comment:
Good question.

I use exceptions so that I can provide context up the call stack about what failed, specifically for debugging purposes.    There are so many filesystem failure possibilities and they can be a pain to debug.  (The alternative was some kind of richer error codes which seemed like overkill).

To prevent a crash, the method that calls this does a try-catch.  See the calling method `suspend fun ExceptionPath.getExceptionDirectory(context: Context): File?`. I think that should address this.